File: backend/src/domain/repositories/device.repository.ts
import { Device } from '../entities/device.entity';

export interface IDeviceRepository {
  getAllDevices(): Promise<Device[]>;
  findByMacAddress(macAddress: string): Promise<Device | null>;
  saveDevice(device: Device): Promise<void>;
  updateLastSeen(macAddress: string, lastSeen: Date): Promise<void>;
}

File: mobile/lib/domain/repositories/device_repository.dart
import '../entities/device.dart';

abstract class IDeviceRepository {
  Future<List<Device>> getAllDevices();
  Future<Device?> getDeviceByMac(String macAddress);
  Future<void> saveDevice(Device device);
  Future<void> updateDeviceStatus(String macAddress, bool isKnown);
}
